Compiling with armclang version 6.17 fails

Running the following command:

```make -f tensorflow/lite/micro/tools/make/Makefile TARGET=cortex_m_corstone_300 TARGET_ARCH=cortex-m55 TOOLCHAIN=armclang BUILD_TYPE=release test_hello_world_test```

with armclang version 6.17 results in the following error:
```
tensorflow/lite/micro/memory_planner/greedy_memory_planner.cc:369:9: error: variable 'memory_use' set but not used [-Werror,-Wunused-but-set-variable]
    int memory_use = 0;
        ^
1 error generated.
tensorflow/lite/micro/tools/make/Makefile:586: recipe for target 'tensorflow/lite/micro/tools/make/gen/cortex_m_corstone_300_cortex-m55_release/obj/core/tensorflow/lite/micro/memory_planner/greedy_memory_planner.o' failed
make: *** [tensorflow/lite/micro/tools/make/gen/cortex_m_corstone_300_cortex-m55_release/obj/core/tensorflow/lite/micro/memory_planner/greedy_memory_planner.o] Error 1
```

--- a/tensorflow/lite/micro/memory_planner/greedy_memory_planner.cc
+++ b/tensorflow/lite/micro/memory_planner/greedy_memory_planner.cc
@@ -366,7 +366,9 @@ void GreedyMemoryPlanner::PrintMemoryPlan() {
     for (int c = 0; c < kLineWidth; ++c) {
       line[c] = '.';
     }
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
     int memory_use = 0;
+#endif
     for (int i = 0; i < buffer_count_; ++i) {
       BufferRequirements* requirements = &requirements_[i];
       if ((t < requirements->first_time_used) ||
@@ -378,7 +380,9 @@ void GreedyMemoryPlanner::PrintMemoryPlan() {
         continue;
       }
       const int size = requirements->size;
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
       memory_use += size;
+#endif
       const int line_start = (offset * kLineWidth) / max_size;
       const int line_end = ((offset + size) * kLineWidth) / max_size;
       for (int n = line_start; n < line_end; ++n) {

